### PR TITLE
Ncnews 7 postcomments

### PR DESCRIPTION
**Pull request by dvbenson**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -101,80 +101,113 @@ describe("APP", () => {
...
+      return request(app)
+        .get("/api/articles/4/comments")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe("This article has no comments yet");
+        });
+    });
+  });
+  describe("POST", () => {
+    test("400: bad body/missing required fields", () => {
```
Comment (Hard-Luck) [inline __tests__/app.test.js:204]: Need a test for if the article id doesnt exist

Comment (Hard-Luck) [reply __tests__/app.test.js:204]: need a test for username doesn't exist

File: controllers/controller.js [hunk 2]
```diff
@@ -33,9 +35,25 @@ const getArticleById = (request, response, next) => {
...
-    .catch(next);
+    .catch((error) => {
+      next(error);
+    });
 };
 
+const postComments = (request, response, next) => {
+  const { article_id: articleId } = request.params;
+  const newComment = request.body;
+  return Promise.all([checkArticleId(articleId), checkNewComment(newComment)])
```
Comment (Hard-Luck) [inline controllers/controller.js:46]: This logic should be handled in the model

**Review by Hard-Luck — CHANGES_REQUESTED** [review]
Just need to move some stuff around and add some more tests

File: __tests__/app.test.js (hunk 1 above)
Comment (dvbenson) [reply __tests__/app.test.js:204]: > Need a test for if the article id doesnt exist

this is in another branch...


> need a test for username doesn't exist

Yep, I agree it is what made this task so difficult. On it.

**Pull request merged**